src/codemodder/codetf/v3/codetf.py

```diff
@@ -148,6 +151,48 @@ def from_v2_aimetadata(ai_metadata: AIMetadatav2) -> AIMetadata:
     )
 
 
+def from_v2_result_per_finding(result: Result) -> FixResult | None:
+    """
+    This transformation assumes that the v2 result will only contain a single fixedFinding for all changesets.
+    """
+    # Find the changeset with a fixedFinding
+    try:
+        changeset: v2ChangeSet = next(cs for cs in result.changeset if cs.fixedFindings)
+    except StopIteration:
+        logger.debug("No fixedFinding in the given Result")
+        return None
+
+    assert changeset.fixedFindings
+    finding = changeset.fixedFindings[0]
+
+    v3changesets = [
+        ChangeSet(
+            path=cs.path, diff=cs.diff, changes=[c.to_common() for c in cs.changes]
+        )
+        for cs in result.changeset
+    ]
+
+    generation_metadata = GenerationMetadata(
+        strategy=Strategy.ai if changeset.ai else Strategy.deterministic,
+        ai=from_v2_aimetadata(changeset.ai) if changeset.ai else None,
+        provisional=False,
```

Comment:
will this function always be called from non-provisional contexts?

Reply:
No clue, not sure why this flag exists in the first place.

Reply:
provisional=True is "magic"... maybe we no longer make this distinction ?

Reply:
It should be derived from the v2 changeset the same way as these other metadata fields.
```suggestion
        provisional=changeset.provisional,
```

Reply:
Is there any chance that for our purposes it makes more sense to pass in these metadata fields as parameters to this new function? For the use case I'm thinking of, I'd argue that the answer is yes.